packages/neon-core/src/tx/transaction/main.ts

```diff
@@ -79,3 +124,67 @@ export function deserializeWitnesses(
   );
   return Object.assign(tx, { scripts });
 }
+
+export function formatSender(sender: string | undefined): string {
+  if (!sender) {
+    return "";
+  }
+  if (sender.length === 42 && sender.startsWith("0x")) {
+    const hex = sender.slice(2);
+    ensureHex(hex);
+    return hex;
+  } else if (sender.length === 40) {
+    ensureHex(sender);
+    return sender;
+  } else if (sender.length === 34) {
+    return getScriptHashFromAddress(sender);
+  } else {
+    throw new Error(`Sender format error: ${sender}`);
+  }
+}
+
+export function getCompressedSize(value: number): number {
+  if (value < 0xfd) {
+    return 1;
+  } else if (value < 0xffff) {
+    return 3;
+  } else {
+    return 5;
+  }
+}
+
+export function getNetworkFeeForSig(): number {

```

Comment:
can we move all the fees related stuff out? Fees can be a different class at this point.

Reply:
OK